enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.ts

```diff
@@ -22,17 +28,47 @@ export const useInitData = ({
   isInitialized: boolean;
 } => {
   const dispatch = useDispatch();
+
   const [isInitialized, setIsInitialized] = useState(false);
   const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [sessionTokenState, setSessionTokenState] =
+    useState<EmbeddingSessionTokenState | null>(null);
 
   useEffect(() => {
     registerVisualizationsOnce();
   }, []);
 
+  useEffect(() => {
+    if (config.authType === "jwt") {
+      const updateToken = () => {
+        const currentState = store.getState();
+        setSessionTokenState(getSessionTokenState(currentState));
+      };
+
+      const unsubscribe = store.subscribe(updateToken);
+
+      if (config.jwtProviderUri) {
+        store.dispatch(
+          getOrRefreshSession(config.jwtProviderUri) as unknown as AnyAction,
```

Comment:
I could really use some help with these types. I've tried just about everything to make the types work (using `ReturnType<store.getState>` for the store, manually setting the type of `createAsyncThunk`, etc) but I can't seem to find a nice solution that doesn't involve ugly casting.